src/test/java/liuni/LiUniResourceTest.java

```diff
@@ -0,0 +1,157 @@
+            assertEquals(respList, resp.getEntity());
+        }
+        catch (Exception e) {
+            Assert.fail("This exception is not expected.");
+        }
+    }
+
+    @Test
+    public void testREST_fetchTimeline_updateTimeline() {
+        ResponseList<Status> respList = new ResponseListImpl<Status>();
+        try {
+            doReturn(respList).when(twitter).getHomeTimeline();
+
+            Response resp = resource.fetchTimeline();
+
+            assertEquals(resp.getStatus(), Response.Status.OK.getStatusCode());
+            assertEquals(resp.getEntity(), respList);
+
+            respList.add(status);
+            resp = resource.fetchTimeline();
+
+            assertEquals(Response.Status.OK.getStatusCode(), resp.getStatus());
+            assertEquals(respList, resp.getEntity());
+        }
+        catch (Exception e) {
+            Assert.fail("This exception is not expected.");
+        }
+    }
+
+    @Test
+    public void testREST_fetchTimeline_fetchTimelineError() {
+        ResponseList<Status> respList = new ResponseListImpl<Status>();
+        respList.add(status);
+        String expectedError = (new ErrorModel()).getGeneralError();
+        try {
+            when(twitter.getHomeTimeline()).thenThrow(new TwitterException("This is an exception test."));
+
+            Response resp = resource.fetchTimeline();
+
+            assertEquals(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode(), resp.getStatus());
+            assertEquals(expectedError, resp.getEntity().toString());
+        }
+        catch (Exception e) {
+            Assert.fail("This exception is not expected.");
+        }
+    }
+
+    /* Test LiUniResource .postTweet() REST method */
+    @Test
+    public void testREST_postTweet_successfulPost() {
+        String testString = "message";
+        try {
+            when(twitter.updateStatus(testString)).thenReturn(status);
+            when(status.getText()).thenReturn(testString);
+
+            Response resp = resource.postTweet(testString);
+
+            assertEquals(Response.Status.CREATED.getStatusCode(), resp.getStatus());
+            assertEquals(testString, resp.getEntity().toString());
+        }
+        catch (Exception e) {
+            Assert.fail("This exception is not expected.");
+        }
+    }
+
+    @Test
+    public void testREST_postTweet_badPost_badTweet() {
+        String testString = "This is a bad string because it is going go to be longer than Twitter's limits. It is this long because it is Lorem Ipsum. Lorem ipsum dolor sit amet, consectetur adipiscing elit. In vehicula velit vel augue porta condimentum. Nullam ornare velit mattis, maximus sem eget, malesuada nibh. Interdum et malesuada fames ac ante ipsum primis in faucibus. Suspendisse potenti. Pellentesque a dictum lorem. Ut sit amet fringilla turpis. Sed pretium, erat tincidunt aliquam mattis, neque diam luctus mauris, sed lacinia arcu libero sit amet tortor.";
```

Comment:
you could also just generate a string based on a set length.  That would be more scalable.,